### PR TITLE
add REST API for userData - backend data fetch powered by GraphQL

### PR DESCRIPTION
**Pull request by TheWillG**
Added an endpoint (/api/v1/github/userData/USER_NAME) to fetch all relevant userData simply by providing userName.
Must provide a githubUserAccessToken inside .env file

### CODE REVIEW

File: server/src/controllers/index.ts [hunk 1]
```diff
@@ -12,5 +14,7 @@ router.post("/user", userValidator.postUser, userController.postUser);
 router.put("/user/:id", userValidator.putUser, userController.putUser);
```
Comment (theblindprophet) [inline server/src/controllers/index.ts]: Let's comment these out for now. I don't see using them soon. 14-15

File: server/src/controllers/githubController.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { getUserData as getData } from "../services/userService";
+
+const getUserData = async (req: Request, res: Response) => {
+  const { userName } = req.params;
+  try {
+    const userData = await getData(userName);
+    res.status(200).send(userData.data.user);
+  } catch (e) {
+    logger.error(`Failed to get github user data with userdata ${userName}`, e);
+    res.status(404).send({});
```
Comment (theblindprophet) [inline server/src/controllers/githubController.ts]: Can we send back the error with the response, maybe `.send({ error: <message> })`

Comment (TheWillG) [reply server/src/controllers/githubController.ts]: Yeah

File: server/src/controllers/index.ts (hunk 1 above)
Comment (TheWillG) [reply server/src/controllers/index.ts]: Yeah all the user stuff is irrelevant for now

**Comment by TheWillG** [comment]
refactored

**Pull request merged**